File: logic/rsp.py
import random  # 导入随机模块以生成随机选择

class RSP:
    def __init__(self):
        # 初始化用户分数和电脑分数
        self.user_score = 0
        self.computer_score = 0

    def get_computer_choice(self):
        # 随机选择电脑的选项（石头、布、剪刀）
        return random.choice(['石头', '布', '剪刀'])

    def determine_winner(self, user_choice, computer_choice):
        # 根据用户和电脑的选择确定胜者
        if user_choice == computer_choice:
            return '平局'  # 如果选择相同，返回平局
        elif (user_choice == '石头' and computer_choice == '剪刀') or \
             (user_choice == '剪刀' and computer_choice == '布') or \
             (user_choice == '布' and computer_choice == '石头'):
            self.user_score += 1  # 用户获胜，增加用户分数
            return '你赢了！'  # 返回用户胜利信息
        else:
            self.computer_score += 1  # 电脑获胜，增加电脑分数
            return '电脑赢了！'  # 返回电脑胜利信息

    def check_winner(self):
        # 检查是否有一方达到胜利条件（分数达到3）
        if self.user_score >= 3:
            winner = '你赢了游戏！'  # 用户胜利信息
            self.reset_scores()  # 重置分数
            return winner  # 返回用户胜利信息
        elif self.computer_score >= 3:
            winner = '电脑赢了游戏！'  # 电脑胜利信息
            self.reset_scores()  # 重置分数
            return winner  # 返回电脑胜利信息
        return None  # 如果没有胜利者，返回 None

    def reset_scores(self):
        # 重置用户和电脑的分数
        self.user_score = 0
        self.computer_score = 0

File: static/js/rsp.js
// 为所有选择按钮添加点击事件监听器
document.querySelectorAll('.choice-button').forEach(button => {
    button.addEventListener('click', function() {
        // 获取用户选择的选项
        const userChoice = this.getAttribute('data-choice');

        // 向服务器发送用户的选择
        fetch('/rsp/play_rsp', {
            method: 'POST', // 使用 POST 方法
            headers: {
                'Content-Type': 'application/json' // 设置请求头为 JSON 格式
            },
            body: JSON.stringify({ choice: userChoice }) // 将用户选择转为 JSON 字符串
        })
        .then(response => response.json()) // 解析服务器返回的 JSON 响应
        .then(data => {
            // 更新结果显示
            document.getElementById('result').innerText = `你选择了 ${userChoice}，电脑选择了 ${data.computer_choice}. ${data.result}`;
            document.getElementById('user-score').innerText = `你: ${data.user_score}`; // 更新用户分数
            document.getElementById('computer-score').innerText = `电脑: ${data.computer_score}`; // 更新电脑分数

            // 更新获胜者信息
            const overallWinnerElement = document.getElementById('overall-winner');
            if (data.overall_winner) { // 如果有获胜者
                overallWinnerElement.innerText = data.overall_winner; // 显示获胜者
                // 根据获胜者改变文本颜色
                if (data.overall_winner.includes('你')) {
                    overallWinnerElement.style.color = 'red'; // 用户获胜时显示红色
                } else {
                    overallWinnerElement.style.color = 'blue'; // 电脑获胜时显示蓝色
                }
            } else {
                overallWinnerElement.innerText = ''; // 清空获胜者信息
                overallWinnerElement.style.color = ''; // 恢复默认颜色
            }
        });
    });
});

// 为重置按钮添加点击事件监听器
document.getElementById('reset-button').addEventListener('click', function() {
    // 发送重置请求到服务器
    fetch('/rsp/reset_rsp', {
        method: 'POST' // 使用 POST 方法
    })
    .then(response => response.json()) // 解析服务器返回的 JSON 响应
    .then(data => {
        // 清空结果和分数显示
        document.getElementById('result').innerText = '';
        document.getElementById('user-score').innerText = '你: 0'; // 重置用户分数
        document.getElementById('computer-score').innerText = '电脑: 0'; // 重置电脑分数
        document.getElementById('overall-winner').innerText = ''; // 清空获胜者信息
    });
});